diff -N -p -r -u linux-2.6.33.4-unmodified/include/drm/drmP.h linux-2.6.33.4/include/drm/drmP.h
--- linux-2.6.33.4-unmodified/include/drm/drmP.h	2010-05-12 15:04:27.000000000 -0700
+++ linux-2.6.33.4/include/drm/drmP.h	2010-05-26 19:22:17.860163428 -0700
@@ -163,6 +163,10 @@ extern void drm_ut_debug_printk(unsigned
 #define DRM_MAX_CTXBITMAP (PAGE_SIZE * 8)
 #define DRM_MAP_HASH_OFFSET 0x10000000
 
+/* for via chrome9 DRM driver */
+#define DRM_MEM_PAGES     9
+#define DRM_MEM_SGLISTS   20
+
 /*@}*/
 
 /***********************************************************************/
